server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java

```diff
@@ -142,25 +142,50 @@ private static void buildComponent(WsCe.Task.Builder builder, @Nullable Componen
     }
   }
 
-  private class ComponentCache {
-    private final DbSession dbSession;
-    private final Map<String, ComponentDto> componentsByUuid = new HashMap<>();
+  private static Set<String> ceQueueDtoToComponentUuids(Iterable<CeQueueDto> dtos) {
+    return from(dtos)
+      .transform(new Function<CeQueueDto, String>() {
+        @Override
+        @Nullable
+        public String apply(@Nonnull CeQueueDto input) {
+          return input.getComponentUuid();
+        }
+      })
+      .filter(notNull())
+      .toSet();
+  }
+
+  private static Set<String> ceActivityDtoToComponentUuids(Iterable<CeActivityDto> dtos) {
+    return from(dtos)
+      .transform(new Function<CeActivityDto, String>() {
+        @Override
+        @Nullable
+        public String apply(@Nonnull CeActivityDto input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)